### PR TITLE
FB-1077 Add timeout for XVerify requests

### PR DESCRIPTION
**Pull request by fauvel**

### CODE REVIEW

**Comment by fauvel** [comment]
Yes, it succeeds. And before we've used Guzzle's defaults, which mean that we had no timeout at all.

File: library/CMService/XVerify/Client.php [hunk 1]
```diff
@@ -5,13 +5,21 @@ class CMService_XVerify_Client extends CM_Service_EmailVerification_Standard {
     /** @var string */
     protected $_domain, $_code;
 
+    /** @var int */
+    protected $_timeout;
+
     /**
-     * @param string $domain
```
Comment (njam) [inline library/CMService/XVerify/Client.php]: This can be confusing. More like "XVerify request failed"?

Comment (fauvel) [inline library/CMService/XVerify/Client.php]: In the end it's wrapped again as an "XVerify client error", is that okay?
```
WARNING XVerify client error
CM_Exception: XVerify timed out in /home/vagrant/cm/library/CMService/XVerify/Client.php on line 76
0. {main} at /home/vagrant/sk/public/index.php line 0
1. CM_Bootloader->execute(Closure) at /home/vagrant/sk/public/index.php line 11
2. {closure}() at /home/vagrant/cm/library/CM/Bootloader.php line 163
3. CM_Http_Handler->processRequest(CM_Http_Request_Post) at /home/vagrant/sk/public/index.php line 9
4. CM_Http_Response_Abstract->process() at /home/vagrant/cm/library/CM/Http/Handler.php line 28
5. CM_Http_Response_View_Abstract->_process() at /home/vagrant/cm/library/CM/Http/Response/Abstract.php line 50
6. CM_Http_Response_Abstract->_runWithCatching(Closure, Closure) at /home/vagrant/cm/library/CM/Http/Response/View/Abstract.php line 190
7. CM_Http_Response_View_Abstract->{closure}() at /home/vagrant/cm/library/CM/Http/Response/Abstract.php line 291
8. CM_Http_Response_View_Ajax->_processView([]) at /home/vagrant/cm/library/CM/Http/Response/View/Abstract.php line 187
9. CM_FormField_Abstract->ajax_validate(['userInput' => 'huih@gmx.de', 'form' => 'SK_Form_SignUp', 'fieldName' => 'email'], CM_Frontend_JavascriptContainer_View, CM_Http_Response_View_Ajax) at /home/vagrant/cm/library/CM/Http/Response/View/Ajax.php line 33
10. CM_FormField_Email->validate(CM_Frontend_Environment, 'huih@gmx.de') at /home/vagrant/cm/library/CM/FormField/Abstract.php line 98
11. CMService_XVerify_Client->isValid('huih@gmx.de') at /home/vagrant/cm/library/CM/FormField/Email.php line 9
12. CMService_XVerify_Client->_getResponseBody('huih@gmx.de') at /home/vagrant/cm/library/CMService/XVerify/Client.php line 34
13. CMService_XVerify_Client->_getResponse('huih@gmx.de') at /home/vagrant/cm/library/CMService/XVerify/Client.php line 87
14. {throw} at /home/vagrant/cm/library/CMService/XVerify/Client.php line 76
Exception meta info:
Array
(
    [messageOriginal] => 'cURL error 28: Resolving timed out after 124 milliseconds (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)'
)
```

Comment (njam) [inline library/CMService/XVerify/Client.php]: I think it's fine. Afaics we're just logging the exception in `_handleException()` as usual, preserving the severify.

Comment (fauvel) [inline library/CMService/XVerify/Client.php]: Yes exactly.

**Pull request merged**